### PR TITLE
style: restructure project

### PR DESCRIPTION
**Pull request by clavin**
This PR restructures the project to use [Yarn Workspaces](https://classic.yarnpkg.com/en/docs/workspaces/) and [TypeScript Project References](https://www.typescriptlang.org/docs/handbook/project-references.html). This keeps each module of the whole project separate and can also speed up builds when you only touch one module (by not having to rebuild the other modules).

### Notes

* When you pull, run both `yarn` and `yarn run build`. The latter is due to [a caveat with TypeScript Project References](https://www.typescriptlang.org/docs/handbook/project-references.html#caveats-for-project-references) and will hopefully be remedied automatically in the future. If you don't run `yarn run build` then you may see errors in your editor for modules that have not been built (or are not yet up-to-date and need to be rebuilt).

### CODE REVIEW

**Review by erickzhao — COMMENTED** [review]
Could we get a root README file on using the workspace structure as well?

**Comment by clavin** [comment]
I've added a barebones section to the README for getting the project up and running, similar to the info in this PR. I'll have to add a more in-depth guide about the workspace structure when I find a less busy moment.

File: README.md [hunk 1]
```diff
@@ -1 +1,13 @@
-# bugbot
\ No newline at end of file
+# BugBot
+
+## Development
+
+BugBot is split into multiple modules, connected in development via [Yarn Workspaces](https://classic.yarnpkg.com/en/docs/workspaces/) and [TypeScript Project References](https://www.typescriptlang.org/docs/handbook/project-references.html). Each module can be found within the `modules/` top-level directory.
+
+After cloning BugBot, run the following commands to setup your workspace:
```
Comment (erickzhao) [inline README.md:7]: small typo:
```suggestion
After cloning BugBot, run the following commands to set up your workspace:
```

We only need to install deps once in the root with Workspaces, right?

**Review by erickzhao — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (clavin) [reply README.md]: You're correct, the yarn workspace will handle installing the deps in the child modules (and deduplicating them too!)

**Review by ckerr — APPROVED** [review]

**Pull request merged**